internal/runners/branch/tree.go

```diff
@@ -0,0 +1,192 @@
+	localBranchFormatting string
+}
+
+const (
+	edgeLink string = "│  "
+	edgeMid  string = "├─"
+	edgeEnd  string = "└─"
+)
+
+func NewBranchTree() *BranchTree {
+	return &BranchTree{tree: make(tree)}
+}
+
+func (bt *BranchTree) BuildFromBranches(branches mono_models.Branches) {
+	bt.rootBranches = getRootBranches(branches)
+	for _, branch := range bt.rootBranches {
+		bt.tree[branchNode{branch.Label, branch.BranchID.String()}] = buildBranchTree(branch, branches)
+	}
+}
+
+func buildBranchTree(currentBranch *mono_models.Branch, branches mono_models.Branches) tree {
+	t := getChildren(currentBranch, branches)
+	for _, branch := range branches {
+		// Discard any branches without tracking information as we are only interested
+		// in child branches of the current branch
+		if branch.Tracks == nil {
+			continue
+		}
+
+		// Check that this branch is a child branch and recursively build its tree
+		if _, ok := t[branchNode{branch.Label, branch.BranchID.String()}]; ok {
+			t[branchNode{branch.Label, branch.BranchID.String()}] = buildBranchTree(branch, branches)
+		}
+	}
+	return t
+}
+
+func getRootBranches(branches mono_models.Branches) mono_models.Branches {
+	var rootBranches mono_models.Branches
+	for _, branch := range branches {
+		// Account for forked projects where the root branches contain
+		// a tracking ID that is not in the current project's branches
+		if branch.Tracks != nil && containsBranch(branch.Tracks, branches) {
+			continue
+		}
+		rootBranches = append(rootBranches, branch)
+	}
+	return rootBranches
+}
+
+func containsBranch(id *strfmt.UUID, branches mono_models.Branches) bool {
+	for _, branch := range branches {
+		if branch.BranchID.String() == id.String() {
+			return true
+		}
+	}
+	return false
+}
+
+func getChildren(branch *mono_models.Branch, branches mono_models.Branches) tree {
+	children := make(tree)
+	if branch == nil {
+		return children
+	}
+
+	for _, b := range branches {
+		if b.Tracks == nil {
+			continue
+		}
+		if b.Tracks.String() == branch.BranchID.String() {
+			children[branchNode{b.Label, b.BranchID.String()}] = make(tree)
+		}
+	}
+	return children
+}
+
+func (bt *BranchTree) AddBranchFormatting(formatting string) {
+	bt.branchFormatting = formatting
+}
+
+func (bt *BranchTree) AddLocalBranch(branch string) {
+	bt.localBranch = branch
+}
+
+func (bt *BranchTree) AddLocalBranchFormatting(formatting string) {
+	bt.localBranchFormatting = formatting
+}
+
+func (bt *BranchTree) String() string {
+	w := new(bytes.Buffer)
+	var levelsCompleted []int
+	bt.print(w, bt.tree, 0, levelsCompleted)
+	return w.String()
+}
+
+func (bt *BranchTree) print(w io.Writer, currentTree tree, currentLevel int, levelsCompleted []int) {
+	// Sort keys at current level to ensure consistent output
+	var nodes []branchNode
+	for k := range currentTree {
+		nodes = append(nodes, k)
+	}
+	sort.Slice(nodes, func(i, j int) bool {
+		return nodes[i].Label < nodes[j].Label
+	})
+
+	for i, node := range nodes {
+		edge := edgeMid
```

Comment:
Can we name `edge` something more descriptive like `prefix` or `listBullet` or something else that doesn't require me to cross reference what this var contains.